tests/spicy/types/type/type-info.spicy

```diff
@@ -0,0 +1,17 @@
+# @TEST-EXEC: spicyc -dj %INPUT >output
+# @TEST-EXEC: btest-diff output
+#
+# @TEST-DOC: Exercises type `type`.
+
+module X;
+
+type S = struct {
+    a: int32;
+    b: int32;
+};
+
+global s1: type = S;
+global s2: type = typeinfo(S);
+global s3: type = typeinfo(bool);
```

Comment:
The LHS is not really a type, but instead a type info, so I am not a fan of calling its type `type` (also easy to confuse with other uses of `type` which currently always introduce new types, either in a type declaration or type alias declaration). Could we call this e.g., `type_info` instead?

Could you add a test for the using `typeinfo` with a value instead of a type, e.g.,

```ruby
print typeinfo(true);
```

I think this will be very useful to users during debugging.

Reply:
> The LHS is not really a type, but instead a type info

Don't agree. The way I see it, is that it *is* a type, it just happens to be represented internally as a `typeinfo`. And from that perspective, the first assignment (s1) is actually what I want: it's assigning a type to a type.

Ideally, we'd then not have the other two versions (s2/s3) and instead directly write:

```
global s3: type = bool;
```

But that's something our parser can't (easily) support, hence the kludge to have type-infos coerce to `type`. 

Note that this PR also documents it that way: `typeinfo(...)` returns a `type`. Maybe `typeinfo` isn't a good name for the operator, but not sure what else to use (`type` would be odd, and probably cause Bison trouble again as well).